paramak/parametric_components/blanket_fp.py

```diff
@@ -155,78 +203,57 @@ def fun(theta):
                 return attribute
         return fun
 
-    def find_points(self):
-        conversion_factor = 2 * np.pi / 360
-
-        def R(theta, pkg=np):
-            """R(theta) plasma profile, theta being the angle in degree
-            """
-            theta *= conversion_factor
-            return self.major_radius + self.minor_radius * pkg.cos(
-                theta + self.triangularity * pkg.sin(theta)
-            )
-
-        def Z(theta, pkg=np):
-            """R(theta) plasma profile, theta being the angle in degree
-            """
-            theta *= conversion_factor
-            return (
-                self.elongation * self.minor_radius * pkg.sin(theta)
-                + self.vertical_displacement
-            )
+    def find_points(self, angles=None):
 
         # create array of angles theta
-        thetas = np.linspace(
-            self.start_angle,
-            self.stop_angle,
-            num=self.num_points,
-            endpoint=True,
-        )
+        if angles is None:
+            thetas = np.linspace(
+                self.start_angle,
+                self.stop_angle,
+                num=self.num_points,
+                endpoint=True,
+            )
+        else:
+            thetas = angles
 
         # create inner points
         inner_offset = self.make_callable(self.offset_from_plasma)
-        inner_points = self.create_offset_points(
-            thetas, R, Z, inner_offset
-        )
+        inner_points = self.create_offset_points(thetas, inner_offset)
         inner_points[-1][2] = "straight"
+        self.inner_points = inner_points
 
         # create outer points
         thickness = self.make_callable(self.thickness)
 
         def outer_offset(theta):
             return inner_offset(theta) + thickness(theta)
 
-        outer_points = self.create_offset_points(
-            np.flip(thetas), R, Z, outer_offset)
+        outer_points = self.create_offset_points(np.flip(thetas), outer_offset)
+        outer_points[-1][2] = "straight"
+        self.outer_points = outer_points
 
         # assemble
         points = inner_points + outer_points
-        points[-1][2] = "straight"
         self.points = points
+        return points
 
-    def create_offset_points(self, thetas, R_fun, Z_fun, offset):
+    def create_offset_points(self, thetas, offset):
         """generates a list of points following parametric equations with an
         offset
 
-        :param thetas: list of angles (radians)
-        :type thetas: list
-        :param R_fun: parametric function for R coordinate (cm)
-        :type R_fun: callable
-        :param Z_fun: parametric function for Z coordinate (cm)
-        :type Z_fun: callable
-        :param offset: offset value (cm). offset=0 will follow the parametric
-         equations.
-        :type offset: callable
-        :return: list of points [[R1, Z1, connection1], [R2, Z2, connection2],
+        Args:
+            thetas (np.array): the angles
```

Comment:
Can we describe these angles a bit more, e.g include the units (degrees or radians)